internal/credit/postgres_connector/grant_test.go

```diff
@@ -0,0 +1,238 @@
+package postgres_connector
+
+import (
+	"context"
+	"log/slog"
+	"testing"
+	"time"
+
+	"entgo.io/ent/dialect"
+	entsql "entgo.io/ent/dialect/sql"
+	"github.com/oklog/ulid/v2"
+	"github.com/peterldowns/pgtestdb"
+	"github.com/stretchr/testify/assert"
+
+	credit "github.com/openmeterio/openmeter/internal/credit"
+	inmemory_lock "github.com/openmeterio/openmeter/internal/credit/inmemory_lock"
+	"github.com/openmeterio/openmeter/internal/credit/postgres_connector/ent/db"
+	"github.com/openmeterio/openmeter/internal/meter"
+	credit_model "github.com/openmeterio/openmeter/pkg/credit"
+	"github.com/openmeterio/openmeter/pkg/models"
+	product_model "github.com/openmeterio/openmeter/pkg/product"
+)
+
+func TestPostgresConnectorGrants(t *testing.T) {
+	namespace := "default"
+	meters := []models.Meter{
+		{
+			Namespace: namespace,
+			ID:        "meter-1",
+			Slug:      "meter-1",
+		},
+	}
+	meterRepository := meter.NewInMemoryRepository(meters)
+	products := []product_model.Product{{
+		Namespace: namespace,
+		MeterSlug: "meter-1",
+		Name:      "product-1",
+	}}
+
+	tt := []struct {
+		name        string
+		description string
+		test        func(t *testing.T, connector credit.Connector, db_client *db.Client)
+	}{
+		{
+			name:        "CreateGrant",
+			description: "Create a grant in the database",
+			test: func(t *testing.T, connector credit.Connector, db_client *db.Client) {
+				ctx := context.Background()
+				p := createProduct(t, connector, namespace, products[0])
+				grant := credit_model.Grant{
+					Subject:     "subject-1",
+					ProductID:   p.ID,
+					Type:        credit_model.GrantTypeUsage,
+					Amount:      100,
+					Priority:    1,
+					EffectiveAt: time.Now().Truncate(0),
+					Expiration: credit_model.ExpirationPeriod{
+						Duration: credit_model.ExpirationPeriodDurationDay,
+						Count:    1,
+					},
+				}
+				g, err := connector.CreateGrant(ctx, namespace, grant)
+				assert.NoError(t, err)
+				// assert count
+				assert.Equal(t, 1, db_client.CreditEntry.Query().CountX(ctx))
+				// assert fields
+				assert.NotNil(t, g.ID)
+				// remove additional fields
+				g.ID = nil
+				assert.Equal(t, g, grant)
+			},
+		},
+		{
+			name:        "VoidGrant",
+			description: "Void a grant in the database and get the latest grant for an ID",
+			test: func(t *testing.T, connector credit.Connector, db_client *db.Client) {
+				ctx := context.Background()
+				p := createProduct(t, connector, namespace, products[0])
+				grant := credit_model.Grant{
+					Subject:     "subject-1",
+					ProductID:   p.ID,
+					Type:        credit_model.GrantTypeUsage,
+					Amount:      100,
+					Priority:    1,
+					EffectiveAt: time.Now().Truncate(0),
```

Comment:
This causes test failures at my side.

```
Error:      	Not equal: 
            	            	expected: credit.Grant{ID:(*string)(nil), ParentID:(*string)(nil), Subject:"subject-1", Type:"USAGE", ProductID:(*string)(0xc00049b470), Amount:100, Priority:0x1, EffectiveAt:time.Date(2024, time.April, 22, 8, 27, 26, 83721000, time.Local), Expiration:credit.ExpirationPeriod{Count:0x1, Duration:"DAY"}, Metadata:map[string]string(nil), Rollover:(*credit.GrantRollover)(nil), Void:false}
            	            	actual  : credit.Grant{ID:(*string)(nil), ParentID:(*string)(nil), Subject:"subject-1", Type:"USAGE", ProductID:(*string)(0xc0000da2d0), Amount:100, Priority:0x1, EffectiveAt:time.Date(2024, time.April, 22, 8, 27, 26, 83721902, time.Local), Expiration:credit.ExpirationPeriod{Count:0x1, Duration:"DAY"}, Metadata:map[string]string(nil), Rollover:(*credit.GrantRollover)(nil), Void:false}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -9,3 +9,3 @@
            	            	  EffectiveAt: (time.Time) {
            	            	-  wall: (uint64) 83721000,
            	            	+  wall: (uint64) 83721902,
            	            	   ext: (int64) 63849371246,
            	Test:       	TestPostgresConnectorGrants/VoidGrant
```

I would rather create  a test helper function called MustParseTime somewhere: 

```

func MustParseTime(tStr string) time.Time {
	parsedTime, err := time.Parse(time.RFC3339, tStr)
	if err != nil {
		panic(err)
	}

	return parsedTime.UTC()
}
```

And then rely on hardcoded times such as:
```
 EffectiveAt: timeutil.MustParseTime("2024-04-22T12:33:00")
```